### PR TITLE
tests: add behavioral coverage for dashboard/utils/charts.jsx

### LINKED ISSUE #1044: Tests: Add test coverage for dashboard/utils/charts.jsx
## Problem
Currently `frontend/src/components/dashboard/utils/charts.jsx` 
has only 12.76% test coverage as found by running `npm run test-coverage`.

Uncovered lines: 19-41, 45-67, 71-93, 97-119, 123-150

## Proposed Solution
Add comprehensive unit tests for all chart components in this file 
to improve coverage significantly.

## Skills
- React Testing Library
- Jest

### PR DESCRIPTION
**Pull request by piyushzgautam99-ship-it**
## Summary
Added behavioral test coverage for all 5 chart components in `frontend/src/components/dashboard/utils/charts.jsx`.

## Related Issues
Closes #1044

## Changes
- Tests cover `componentsFn` factory logic for all 5 chart components
- dataKey, fill, stroke assertions for Area components
- 7 edge cases for `FeedsTypesChart` (null, undefined, single-entry, multi-entry, stackId, colors, date field skipping)
- `accessorFnAggregation` identity function verified for all charts
- URL wiring verified for all charts

## Type of change
- [x] Chore (test coverage improvement)

## Checklist
- [x] I have read and understood the rules about how to Contribute to this project
- [x] My branch is based on `develop`
- [x] The pull request is for the branch `develop`
- [x] I have added tests for the feature/bug I solved
- [x] All the tests gave 0 errors
- [x] I have reviewed and verified any LLM-generated code included in this PR

## GUI changes
N/A

### CODE REVIEW

**Comment by piyushzgautam99-ship-it** [comment]
Hi @regulartim  I've opened this fresh PR as you suggested. This one has only 1 commit and 1 file changed (charts.test.jsx) — no package-lock.json changes. Please let me know if any further improvements are needed!

File: frontend/tests/components/dashboard/utils/charts.test.jsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+const { mockAnyChartWidget } = vi.hoisted(() => ({
+  mockAnyChartWidget: vi.fn(),
+}));
+
+vi.mock("@certego/certego-ui", () => ({
+  AnyChartWidget: mockAnyChartWidget,
+  getRandomColorsArray: () => Array(30).fill("#aabbcc"),
+}));
+
+vi.mock("../../../../constants/api", () => ({
```
Comment (armoredvortex) [inline frontend/tests/components/dashboard/utils/charts.test.jsx]: are you sure this is correct?
I think it should be `../../../../src/constants/api`

File: frontend/tests/components/dashboard/utils/charts.test.jsx [hunk 2]
```diff
@@ -0,0 +1,229 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
```
Comment (armoredvortex) [inline frontend/tests/components/dashboard/utils/charts.test.jsx]: screen is unused.

**Comment by piyushzgautam99-ship-it** [comment]
Fixed! Corrected the import paths to ../../../../src/constants/api and ../../../../src/constants, and removed the unused screen import. Thanks for the review

**Comment by piyushzgautam99-ship-it** [comment]
Hi @regulartim  totally agree! I'll merge my additional edge case tests into the existing frontend/tests/components/dashboard/charts.test.jsx file instead of maintaining a separate one. I'll update the PR shortly

**Comment by piyushzgautam99-ship-it** [comment]
Updated! Merged the edge cases into the existing tests/components/dashboard/charts.test.jsx and removed the duplicate utils/charts.test.jsx. Added tests for null/undefined data, correct dataKeys for all chart types, and first-element-only reading for FeedsTypesChart.
